soccer/gameplay/skills/goalside_mark.py

```diff
@@ -0,0 +1,163 @@
+import role_assignment
+
+
+class Goalside_Mark(single_robot_behavior.SingleRobotBehavior):
+    '''
+    The goalside mark will take a robot or mark point and position itself on line with the best shot from 
+    the position of the robot/mark_point.
+    @param: ratio - this will control how far from the opposing robot the defender will mark (0 is close to robot)
+    '''
+
+    def __init__(self):
+        super().__init__(continuous=True)
+        #Below params are described above @properties
+        self._ratio = 0.2
+        #This threshold is lower than mark to ensure we get goal side of the opponent fast, 
+        #then back off based on ratio
+        
+        self._mark_line_thresh = constants.Robot.Radius*4
+        self._mark_robot = None 
+        self._mark_point = None 
+
+        self._target_point = None 
+
+        self.add_transition(behavior.Behavior.State.start,
+                            behavior.Behavior.State.running, lambda: True,
+                            "immediately")
+        self.kick_eval = robocup.KickEvaluator(main.system_state())
+        self.mark_pos = None
+        self.adjusted_mark_pos = None
+
+    def execute_running(self):
+        #pylint: disable=no-member
+        #Skill does nothing if mark point isn't given AND the ball or robot to mark can't be found
+        if self.mark_point is None and \
+           (self.mark_robot is None or
+            not main.ball().valid or
+            not self.mark_robot.visible):
+            return
+        
+        #Finds the line from the mark position to the shot point and creates a line between them
+        #removing the overlap with the ball on one side and robot on the other
+        #This assumes even with mark position parameter that there is a robot there to avoid
+        self._reset_mark_pos()
+        mark_line, shot_pt = self.get_mark_segment()
+
+        #Drawing for simulator 
+        main.system_state().draw_line(mark_line, (0, 0, 255), "Mark")
+
+        #Distance from robot to mark line
+        mark_line_dist = mark_line.dist_to(self.robot.pos)
+
+        #Sets target point to nearest point on mark line if the robot is over ball_mark_threshold
+        #from the mark line
+        # or
+        #Sets target point to place on line defined by ratio- 
+        #   - 0 being close to ball, 1 close to mark pt
+        self._target_point = None
+        if mark_line_dist > self.mark_line_thresh:
+            self._target_point = mark_line.nearest_point(self.robot.pos)
+        else:
+            self._target_point = self.adjusted_mark_pos - (
+                self.mark_pos -
+                shot_pt).normalized() * self.ratio * mark_line.length()
+
+        #Drawing for simulator
+        main.system_state().draw_circle(self.mark_pos, constants.Robot.Radius * 1.2,
+                                        (0, 127, 255), "Mark")
+
+        #Move robot into position and face the ball
+        self.robot.move_to(self._target_point)
```

Comment:
This should be a move subbehavior since its there. `SingleRobotCompositeBehavior` allows for a single subbehavior

Reply:
This is not a composite behavior - I should change that?

Reply:
Eh, I was looking at some of the other skills some more and it's like 50-50 whether they do this. It's up to you